cityofaustin/atd-moped
# Backfill projects with template milestones

## Associated issues
This will resolve https://github.com/cityofaustin/atd-data-tech/issues/9257 once it's approved and tested in staging and run against production.

## Testing
**URL to test:** Local

**Steps to test:**
1. You'll need a python 3.x environment with `requests` installed.
2. Use the Moped editor to create a new project. Do not use the milestone template when you create the project.
3. Edit the project type of your project to one of "Signal - Mod", "Signal - New", "PHB - Mod", "PHB - New"
4. Run `create_milestones.py`, and set `--env (-e)` to `local` and the `max-date-added (-d)` arg date to be in the future

```
python create_milestones.py -e local -d '2022-06-17 00:00:00'
```

5. Navigate to your project timeline can confirm the template milestones have been added.

Other things you can test:
- Repeat the above steps, but use a milestone template from the new project workflow. Run the script. No milestones should be added to the project.
- Repeat the above steps, but set the project type to "Fiber - Mod". Run the script. No milestones should be added to the project.
- Repeat the above steps, but add a "Completed" phase to your project and set it as the current phase. Run the script. Your project should **not** have milestones added. Completed and post-construction projects are excluded.

---
#### Ship list
- [ ] Run against staging environment
- [ ] Run against production environment
